### PR TITLE
Add a timestamp and Git commit to the dartdoc home page

### LINKED ISSUE #7974: docs.flutter.io should show git hash
That will make it easier to verify if the dart docs are upto date.

Maybe also include a "last updated on: <date>".

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/7974

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -90,6 +94,20 @@ dependencies:
   createIndexAndCleanup();
 }
 
+void createFooter(String footerPath) {
+  ProcessResult gitResult = Process.runSync('git', <String>['rev-parse', 'HEAD']);
+  if (gitResult.exitCode != 0)
+    exit(gitResult.exitCode);
```
Comment (Hixie) [inline dev/tools/dartdoc.dart]: does a message get output by git in this case? maybe we should print something to the console here just to make it easier for us to figure out what went wrong

File: dev/tools/pubspec.yaml [hunk 2]
```diff
@@ -5,4 +5,5 @@ dependencies:
   archive: ^1.0.20
   args: ^0.13.4
   http: ^0.11.3
+  intl: '>=0.14.0 <0.15.0'
```
Comment (abarth) [inline dev/tools/pubspec.yaml]: You can write this more compactly as `^0.14.0`

**Comment by abarth** [comment]
LGTM

File: dev/tools/dartdoc.dart [hunk 3]
```diff
@@ -90,6 +94,20 @@ dependencies:
...
+  String gitHead = gitResult.stdout.trim();
+
+  String timestamp = new DateFormat('yyyy-MM-dd HH:mm').format(new DateTime.now());
+
+  new File(footerPath).writeAsStringSync(
+    '<p class="text-center" style="font-size: 10px">'
```
Comment (Hixie) [inline dev/tools/dartdoc.dart]: there's a `<footer>` HTML element for this purpose, fwiw.

**Comment by Hixie** [comment]
LGTM

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (jason-simmons) [inline dev/tools/dartdoc.dart]: Changed this to use a placeholder instead of exiting if git fails

File: dev/tools/pubspec.yaml (hunk 2 above)
Comment (jason-simmons) [inline dev/tools/pubspec.yaml]: done

File: dev/tools/dartdoc.dart (hunk 3 above)
Comment (jason-simmons) [inline dev/tools/dartdoc.dart]: The dartdoc generated page already contains a <footer>.  This text will be placed above that.

**Comment by sethladd** [comment]
Thanks!

**Pull request merged**